activerecord/lib/active_record/relation/query_attribute.rb

```diff
@@ -36,10 +36,27 @@ def unboundable?
         @_unboundable
       end
 
```

Comment:
I wondered whether to place the patch in the base `ActiveModel::Attribute` class or in `ActiveRecord::Relation::QueryAttribute`. I went with the former to keep the dup overhead only for the case where we need it, just as it was in the original [patch](https://github.com/rails/rails/commit/5abb45d53ae32f9d80fac5a21fe4cda9b83412a1#diff-7743ab95b8e15581f432206245c691434a3993d1a751b9d451170956d59457a9R8).